hazelcast-jet-core/src/main/java/com/hazelcast/jet/JetInstance.java

```diff
@@ -306,6 +320,42 @@ default JobStateSnapshot getJobStateSnapshot(@Nonnull String name) {
     @Nonnull
     JetCacheManager getCacheManager();
 
+    /**
+     * Returns an {@link Observable} instance with the specified name.
+     * Represents a flowing sequence of events produced by jobs containing
+     * {@link com.hazelcast.jet.pipeline.Sinks#observable(String)
+     * observable sinks}.
+     * <p>
+     * You can acquire many instances of the same {@code Observable}. They will
+     * all produce the same data.
+     * <p>
+     * In order to observe the events, register an {@link Observer}s on the
+     * {@code Observable}.
+     *
+     * @param name name of the observable
+     * @return observable with the specified name
+     *
+     * @since 4.0
+     */
+    @Nonnull
+    <T> Observable<T> getObservable(@Nonnull String name);
+
+    /**
+     * Returns a new observable with a randomly generated name
+     *
+     * @since 4.0
+     */
+    @Nonnull
+    default <T> Observable<T> newObservable() {
+        return getObservable(UuidUtil.newUnsecureUuidString());
+    }
+
+    /**
+     * Returns a list of all the observables that are active
+     */
+    @Nonnull
+    List<Observable> getObservables();
```

Comment:
no test coverage

Reply:
Will add. Question though: why did you make this return a `List`? Since there is no ordering whatsoever, wouldn't it make more sense to use a `Set`? Ie. `List` is misleading a bit, at least seems that way to me.

Reply:
We can also use `Collection`